src/Ursa/Controls/TimeBox.cs

```diff
@@ -390,18 +388,18 @@ private void ParseTimeSpan(bool showLeadingZero, bool skipParseFromText = false)
 
         if (!skipParseFromText)
         {
-            _values[0] = int.TryParse(_hourText.Text, out int hour) ? hour : 0;
-            _values[1] = int.TryParse(_minuteText.Text, out int minute) ? minute : 0;
-            _values[2] = int.TryParse(_secondText.Text, out int second) ? second : 0;
-            _values[3] = int.TryParse(_milliSecondText.Text, out int millisecond) ? millisecond : 0;
+            _values[0] = int.TryParse(_hourText?.Text, out int hour) ? hour : 0;
+            _values[1] = int.TryParse(_minuteText?.Text, out int minute) ? minute : 0;
+            _values[2] = int.TryParse(_secondText?.Text, out int second) ? second : 0;
+            _values[3] = int.TryParse(_milliSecondText?.Text, out int millisecond) ? millisecond : 0;
         }
 
         VerifyTimeValue();
 
-        _hourText.Text = _values[0].ToString(format);
-        _minuteText.Text = _values[1].ToString(format);
-        _secondText.Text = _values[2].ToString(format);
-        _milliSecondText.Text = _values[3].ToString(format);
+        _hourText?.SetValue(TextPresenter.TextProperty,_values[0].ToString(format));
```

Comment:
By your code we should use SetValue here, but I doubt if we should use SetValue or SetCurrentValue. please help to confirm, or leave it if you don't know.